### PR TITLE
feat: new table function `set_cache_capacity`

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

- introduce new table function `system$set_cache_capacity`
~~~
mysql> call system$set_cache_capacity('memory_cache_block_meta', 1000);
+------------------------+--------+
| node                   | result |
+------------------------+--------+
| VV1M1Mrr5x05UuIY0nUjT  | Ok     |
| ZN4rq5TfxyWBI1cH6KfRR6 | Ok     |
| RNIHu27BbkIv41GxjZ3t27 | Ok     |
+------------------------+--------+
~~~
-  Add block meta cache, capacity set to 0 by default
   during pruning, block meta cache will be used if presents
- tweak `system.caches` table, add new columns `capacity`, `unit`, `access`, `hit`, `miss`

~~~
+------------------------+--------------------------------------------+-----------+------+------------+-------+--------+------+------+
| node                   | name                                       | num_items | size | capacity   | unit  | access | hit  | miss |
+------------------------+--------------------------------------------+-----------+------+------------+-------+--------+------+------+
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_table_snapshot                |         1 |    1 |        256 | count |      2 |    2 |    0 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_table_statistics              |         0 |    0 |        256 | count |      0 |    0 |    0 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_compact_segment_info          |         1 |  552 | 1073741824 | bytes |      1 |    1 |    0 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_bloom_index_filter            |         0 |    0 | 2147483648 | bytes |      0 |    0 |    0 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_bloom_index_file_meta_data    |         0 |    0 |       3000 | count |      0 |    0 |    0 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_block_meta                    |         1 |    1 |       3000 | count |      1 |    0 |    1 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_inverted_index_file_meta_data |         0 |    0 |       3000 | count |      0 |    0 |    0 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_inverted_index_file           |         0 |    0 | 2147483648 | bytes |      0 |    0 |    0 |
| Vq3IAEgM0XxGcJC3S4LUH7 | memory_cache_parquet_file_meta             |         0 |    0 |       3000 | count |      0 |    0 |    0 |
+------------------------+--------------------------------------------+-----------+------+------------+-------+--------+------+------+
~~~

- `minitrace::trace` removed from range index `apply`, it hurts performance (a lot)

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16016)

### CODE REVIEW

File: src/query/service/src/interpreters/access/privilege_access.rs [hunk 1]
```diff
@@ -520,6 +523,34 @@ impl PrivilegeAccess {
         Ok(())
     }
 
+    async fn validate_table_function_access(&self, table_func_name: &str) -> Result<()> {
+        if SYSTEM_TABLE_FUNCTIONS.iter().any(|x| x == &table_func_name) {
+            // need Super privilege to invoke system table functions
+            let privilege = UserPrivilegeType::Super;
+            let session = self.ctx.get_current_session();
+            let current_user = self.ctx.get_current_user()?;
+            session
```
Comment (flaneur2020) [inline src/query/service/src/interpreters/access/privilege_access.rs:532]: just `session.validate_privilege(&GrantObject::Global, privilege, true).await` ?

i suppose that the error message has already been formated as the blow inside this `validate_privilege` function

**Comment by flaneur2020** [comment]
LGTM, @TCeason PTAL

File: src/query/service/src/interpreters/access/privilege_access.rs (hunk 1 above)
Comment (dantengsky) [reply src/query/service/src/interpreters/access/privilege_access.rs:532]: The "default" error message is `ERROR 1105 (HY000): PermissionDenied. Code: 1063, Text = Permission denied.`, which is not very informative, IMO.

**Comment by TCeason** [comment]
> LGTM, @TCeason PTAL
> 
> 
> 
> about the privlege checking part: https://github.com/datafuselabs/databend/pull/16016/commits/fff73110503e13dcf2263d1514f15791ccdba1d3

LGTM, but I'm not sure only this func need check super privilege?

**Review by TCeason — APPROVED** [review]
LGTM. And other table function maybe need to do same process. Let's do it in other pr.

**Pull request merged**